packages/vite/src/node/plugins/optimizedDeps.ts

```diff
@@ -67,7 +70,7 @@ export function optimizedDepsPlugin(config: ResolvedConfig): Plugin {
         // load hooks to avoid race conditions, once processing is resolved,
         // we are sure that the file has been properly save to disk
         try {
-          return await fs.readFile(file, 'utf-8')
+          return loadOptimizedDep(file, depsOptimizer)
```

Comment:
Could we stream .contents, or use them here?

Reply:
I think we need to make it a string here. For sourcemaps, since they're handled in the transform middleware, we could, but I tried that and it doesn't work since etags are generated via the `Buffer` instance, and `Buffer.isBuffer(new Uint8Array(2))` is false.